### PR TITLE
Hadoop 18073 changelog 2

### PR DESCRIPTION
**Pull request by ahmarsuhail**
Updates changelog

### CODE REVIEW

File: hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+|setSocketTimeout    |httpClientBuilder.socketTimeout    |
+|setRequestTimeout    |overrideConfigBuilder.apiCallAttemptTimeout    |
+|setSocketBufferSizeHints    |Not supported    |
+|setSignerOverride    |Not done yet    |
+|setProxyHost    |proxyConfigBuilder.endpoint    |
+|setProxyPort    |set when setting proxy host with .endpoint    |
+|setProxyUsername    |proxyConfigBuilder.username    |
+|setProxyPassword    |proxyConfigBuilder.password    |
+|setProxyDomain    |proxyConfigBuilder.ntlmDomain    |
+|setProxyWorkstation    |proxyConfigBuilder.ntlmWorkstation    |
```
Comment (passaro) [inline hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md]: ntlmDomain and ntlmWorkstation are not supported on the "async" netty.ProxyConfiguration.Builder

File: hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+* [DeleteObject](#deleteobject)
+* [CredentialsProvider](#credentialsprovider)
+* [Auditing](#auditing)
+* [Metric Collection](#metric-collection)
+* [Exception Handling](#exception-handling)
+* [Failure Injection](#failure-injection)
+
+### Client Configuration:
+
+ * We now have two clients, a sync S3 Client and an async S3 Client. The async s3 client is required as the select operation is currently only supported on the async client. Once we are confident in the current set of changes, we will also be exploring moving other operations to the async client as this could provide potential performance benefits.
```
Comment (passaro) [inline hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md]: Maybe make clearer the further changes are out of scope for this PR

File: hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+withForceGlobalBucketAccessEnabled(true) which will allow access to buckets not in this region too.
+Since the SDK V2 no longer supports cross region access, we need to set the region and endpoint of
+the bucket. The behaviour has now been changed to:
+
+* If no endpoint is specified, use s3.amazonaws.com.
+* When setting the endpoint, also set the protocol (HTTP or HTTPS)
+* When setting the region, first initiate a default S3 Client with region eu-west-2. Call headBucket
+  using this client. If the bucket is also in eu-west-2, then this will return a successful
+  response. Otherwise it will throw an error with status code 301 permanently moved. This error
+  contains the region of the bucket in it’s header, which we can then use to configure the client.
```
Comment (passaro) [inline hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md]: it's -> its

File: hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+  * contentLength is not set, as this is a system defined header, and copied over automatically by
+    S3 during copy.
+* `HeaderProcessing.cloneObjectMetadata`: This was previously also setting a lot of system defined
+  metadata, eg: `setHttpExpiresDate` and `setLastModified`. These have been removed as they are set
+  by S3 during the copy. Have tested, and can see they are set automatically regardless of the
+  metadataDirective (copy or replace).
+* `RequestFactoryImpl. copyEncryptionParameters()` : Due to the changes
+  in `EncryptionSecretOperations`, source and destination encryption params have to be set manually.
+
+### MutlipartUpload
```
Comment (passaro) [inline hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md]: typo

File: hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+
+### MultipartUpload
+
+* `RequestFactoryImpl.newObjectMetdata()` : Metadata is now set on the request builder. For MPU, only
+content encoding needs to be set, as per per previous behaviour. Encryption params are set
+in ` multipartUploadEncryptionParameters`.
+
+### GetObject
+
+* Previously, `S3AInputStream` had to keep a reference to the `S3Object` while holding the wrapped
```
Comment (ahmarsuhail) [inline hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md:196]: @passaro could you also mention that the GetObject response has changed, and it now returns the inputstream directly. and maybe also how this differs from V1 (no need to do getWrappedStream anymore I think??)

**Review by passaro — APPROVED** [review]

File: hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+In SDK v1, bulk delete would throw a `com.amazonaws.services.s3.model.MultiObjectDeleteException`
+in case of partial failure. In v2, instead, it returns a `DeleteObjectsResponse` containing a
+list of errors. A new `MultiObjectDeleteException` class was introduced in
+`org.apache.hadoop.fs.s3a` and is thrown when appropriate to reproduce the previous behaviour.
+* `MultiObjectDeleteSupport.translateDeleteException` was moved into `MultiObjectDeleteException`.
+* `ObjectIdentifier` replaces DeleteObjectsRequest.KeyVersion.
+
+### Select
+
+In SDK v2, Handling of select requests has changes significantly since SelectObjectContent is
```
Comment (ahmarsuhail) [inline hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md:215]: Unsure of how much detail to go in here. I think this looks good, but once we're happy with the implementation we'll probably want to add more info. For now should we add some more details about how publisher/subscriber are working, more specifically mentioning lazy loading

**Review by ahmarsuhail — COMMENTED** [review]
one minor change for GetObject and a suggestion for select
